app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadService.java

```diff
@@ -376,6 +378,7 @@ private void updateForeground() {
         long downloaded = Long.parseLong(chunk.getRangeHeader().split("-")[0]);
         if (fullFile.exists() && fullFile.length() == chunk.getSize()) {
           // Mark chunk status as downloaded
+          filesList.add(fullFile.getPath());
```

Comment:
This isn't a workable solution because the Service can be killed and restarted by Android/ the user. filesList will not persist.